editor/src/com/talosvfx/talos/editor/addons/bvb/PropertiesPanel.java

```diff
@@ -26,17 +31,52 @@ public PropertiesPanel(IPropertyProvider propertyProvider, Skin skin) {
         titleLabel = new Label("", getSkin());
         titleLabel.setEllipsis(true);
         Table titleContainer = new Table();
-        titleContainer.add(titleLabel).padTop(-titleLabel.getPrefHeight()-3).growX().left().width(0).row();
+        titleContainer.add(titleLabel).padTop(-titleLabel.getPrefHeight()-3).growX().left();
+
+        ImageButton settingButton = new ImageButton(TalosMain.Instance().getSkin().getDrawable( "icon-edit"));
+        settingButton.setSize(17,17);
+        titleContainer.add(settingButton).padTop(-titleLabel.getPrefHeight()-3).right().row();
+
+        ContextualMenu contextualMenu = new ContextualMenu();
+
```

Comment:
Why not just check if the PropertyProivder is of type Component? Then we don't have to implement those methods in every component